src/istio-watcher/pkg/watcher/watcher.go

```diff
@@ -181,6 +182,12 @@ func (m *IstioWatcher) CollectIstioConnectionMetrics(ctx context.Context, namesp
 }
 
 func (m *IstioWatcher) getEnvoyMetricsFromSidecar(ctx context.Context, pod corev1.Pod, metricsChan chan<- *EnvoyMetrics) error {
+	if !lo.ContainsBy(pod.Spec.Containers, func(item corev1.Container) bool {
+		return item.Name == IstioSidecarContainerName
+	}) {
+		return nil
+	}
+
```

Comment:
I really prefer this with `exists :=.. if !exists` for clarity, but I guess it saves a line.. w/e you prefer.

Reply:
moved into function for clarity, wdyt?

Reply:
Cool as a :cucumber: